app/src/main/java/app/dto/servicedefinition/UpdateServiceDefinitionAttributeDTO.java

```diff
@@ -12,53 +12,56 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package app.model.service.servicedefinition;
+package app.dto.servicedefinition;
 
+import app.model.service.AttributeDataType;
 import com.fasterxml.jackson.annotation.JsonProperty;
 import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlElementWrapper;
 import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlProperty;
 import io.micronaut.core.annotation.Introspected;
 
+import javax.validation.constraints.Min;
+import javax.validation.constraints.NotBlank;
 import java.util.List;
 
 @Introspected
-public class ServiceDefinitionAttribute {
+public class UpdateServiceDefinitionAttributeDTO {
 
     private String code;
-    private boolean variable;
     private AttributeDataType datatype;
-    private boolean required;
+    private Boolean variable;
+    private Boolean required;
     private String description;
 
+    @Min(1)
     @JsonProperty("order")
-    private int attributeOrder;
+    private Integer attributeOrder;
```

Comment:
I'm curious why primitives can't be used here anymore?

Reply:
For a Patch request, if the attribute is null, then we do not update the property in the database.

Reply:
Makes sense :+1: